### PR TITLE
Implement logging for behavioral cloning

### PR DESCRIPTION
**Pull request by sivral**
## Checklist
- [x] All TODOs in the code have been resolved or linked to a proper issue.
- [x] Code has been (auto)formatted.
- [x] Documentation (e.g., README, CHANGELOG, Wiki) has been updated.
- [x] All automated checks have passed.

### CODE REVIEW

File: aegis_gym/rsl/behavior_cloning.py [hunk 1]
```diff
@@ -1,25 +1,46 @@
 import os
 import time
+from pathlib import Path
 from collections import deque
 from collections.abc import Iterator
```
Comment (macmacal) [inline aegis_gym/rsl/behavior_cloning.py:5]:
```suggestion
from collections import deque
from collections.abc import Iterator
from pathlib import Path
```

File: aegis_gym/rsl/behavior_cloning.py [hunk 2]
```diff
@@ -1,25 +1,46 @@
...
 
 import numpy as np
 import torch as th
 import torch.nn as nn
 import torch.nn.functional as F
-from torch.utils.tensorboard import SummaryWriter
+from clearml import Task
+
+from rsl_rl.utils.logger import Logger
```
Comment (macmacal) [inline aegis_gym/rsl/behavior_cloning.py:12]:
```suggestion
from clearml import Task
from rsl_rl.utils.logger import Logger
```

File: aegis_gym/rsl/behavior_cloning.py [hunk 3]
```diff
@@ -1,25 +1,46 @@
...
+        self, env, cfg: dict, teacher: nn.Module, log_dir: Path, device: str = "cpu"
+    ):
         self._env = env
         self._cfg = cfg
         self._device = device
         self._teacher = teacher
         self._num_steps_per_env = cfg["num_steps_per_env"]
 
+        if Task.current_task():
+            Task.current_task().close()
```
Comment (macmacal) [inline aegis_gym/rsl/behavior_cloning.py:32]: Why we need this logic in the first place?

File: aegis_gym/rsl/behavior_cloning.py [hunk 4]
```diff
@@ -120,19 +139,17 @@ def learn(self, num_learning_iterations: int, log_dir: str) -> None:
...
-                tf_writer.add_scalar("loss/pose_loss", avg_pose_loss, it)
-                tf_writer.add_scalar(
-                    "loss/total_loss", avg_action_loss + avg_pose_loss, it
+                self.logger.writer.add_scalar("Loss/action", avg_action_loss.item(), it)
+                self.logger.writer.add_scalar("Loss/pose", avg_pose_loss.item(), it)
+                self.logger.writer.add_scalar(
+                    "Loss/total", (avg_action_loss + avg_pose_loss).item(), it
                 )
-                tf_writer.add_scalar("lr", current_lr, it)
-                tf_writer.add_scalar("buffer_size", self._buffer.size, it)
```
Comment (macmacal) [inline aegis_gym/rsl/behavior_cloning.py:132]: What about the learning rate (`Train/learning_rate` and buffer size (`Train/buffer_size`) scalars?

**Review by macmacal — CHANGES_REQUESTED** [review]

File: aegis_gym/rsl/behavior_cloning.py [hunk 5]
```diff
@@ -99,10 +121,10 @@ def learn(self, num_learning_iterations: int, log_dir: str) -> None:
                 # Backward pass
                 self._optimizer.zero_grad()
                 total_loss.backward()
-                self._optimizer.step()
                 th.nn.utils.clip_grad_norm_(
                     self._policy.parameters(), self._cfg["max_grad_norm"]
                 )
+                self._optimizer.step()
```
Comment (macmacal) [inline aegis_gym/rsl/behavior_cloning.py:127]: How it worked before without the grad norm clipping?

File: aegis_gym/rsl/behavior_cloning.py [hunk 6]
```diff
@@ -146,14 +170,19 @@ def learn(self, num_learning_iterations: int, log_dir: str) -> None:
                 info_str += f" | FPS:           {int(fps)}"
                 print(info_str)
 
-                if len(self._rewbuffer) > 0:
-                    tf_writer.add_scalar("reward/mean", np.mean(self._rewbuffer), it)
+                if self.logger is not None and len(self._rewbuffer) > 0:
+                    self.logger.writer.add_scalar(
+                        "Reward/mean", np.mean(self._rewbuffer), it
+                    )
```
Comment (macmacal) [inline aegis_gym/rsl/behavior_cloning.py]: Its is messy, more lines are required for the logging than for the training logic.

I insist to extract the logging functionality to an auxiary method. There will be probably a need to create 1-3 such functions.

**Review by macmacal — CHANGES_REQUESTED** [review]

File: aegis_gym/rsl/behavior_cloning.py (hunk 5 above)
Comment (macmacal) [reply aegis_gym/rsl/behavior_cloning.py:127]: The original repo probably have a bug: https://github.com/Genesis-Embodied-AI/Genesis/blob/67e3c8d9744a1a4ac66a5359ba7c08ea90084fb0/examples/manipulation/behavior_cloning.py#L88

**Review by macmacal — APPROVED** [review]
LGTM 🆗
